app/src/main/java/org/liberty/android/fantastischmemo/ui/RecentListFragment.java

```diff
@@ -303,24 +325,29 @@ public void onBindViewHolder(final ViewHolder holder, final int position) {
             holder.itemView.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    Intent myIntent = new Intent();
-                    myIntent.setClass(context, StudyActivity.class);
-                    String dbPath = currentItem.dbPath;
-                    myIntent.putExtra(StudyActivity.EXTRA_DBPATH, dbPath);
-                    recentListUtil.addToRecentList(dbPath);
-                    context.startActivity(myIntent);
+                    if (getSelectedItemCount() == 0) {
+                        Intent myIntent = new Intent();
+                        myIntent.setClass(context, StudyActivity.class);
+                        String dbPath = currentItem.dbPath;
+                        myIntent.putExtra(StudyActivity.EXTRA_DBPATH, dbPath);
+                        recentListUtil.addToRecentList(dbPath);
+                        context.startActivity(myIntent);
+                    } else {
+                        toggleSelection(position);
+                    }
+                    updateActionMode();
                 }
             });
 
             holder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
                 @Override
                 public boolean onLongClick(View v) {
-                    String dbPath = currentItem.dbPath;
-                    DialogFragment df = new OpenActionsFragment();
-                    Bundle b = new Bundle();
-                    b.putString(OpenActionsFragment.EXTRA_DBPATH, dbPath);
-                    df.setArguments(b);
-                    df.show(((FragmentActivity)context).getSupportFragmentManager(), "OpenActions");
+                    AppCompatActivity appCompatActivity = (AppCompatActivity) context;
```

Comment:
I agree the second approach is better. The easier approach is using green robot eventbus instead LocalBroadcastManager. I am ok adding it as a dependency. http://greenrobot.org/eventbus/